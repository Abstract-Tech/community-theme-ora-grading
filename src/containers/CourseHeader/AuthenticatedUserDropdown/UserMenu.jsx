import React from 'react';
import PropTypes from 'prop-types';

import { getConfig } from '@edx/frontend-platform';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { Dropdown } from '@edx/paragon';

import messages from '../messages';

export class UserMenu extends React.Component {
  menuItem(href, message) {
    return (
      <Dropdown.Item href={href}>
        {this.props.intl.formatMessage(message)}
      </Dropdown.Item>
    );
  }

  render() {
    const { username } = this.props;
    const { LMS_BASE_URL, LOGOUT_URL } = getConfig();
    return (
      <Dropdown.Menu className="dropdown-menu-right">
        {this.menuItem(`${LMS_BASE_URL}/dashboard`, messages.dashboard)}
        {this.menuItem(`${LMS_BASE_URL}/u/${username}`, messages.profile)}
        {this.menuItem(`${LMS_BASE_URL}/account/settings`, messages.account)}
        {this.menuItem(LOGOUT_URL, messages.signOut)}
      </Dropdown.Menu>
    );
  }
}

UserMenu.propTypes = {
  intl: intlShape.isRequired,
  username: PropTypes.string.isRequired,
};

UserMenu.defaultProps = {};

export default injectIntl(UserMenu);
