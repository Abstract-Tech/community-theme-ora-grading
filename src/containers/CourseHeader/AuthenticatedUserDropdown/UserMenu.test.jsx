import React from 'react';
import { shallow } from 'enzyme';

import { UserMenu } from './UserMenu';

jest.mock('@edx/frontend-platform', () => ({
  getConfig: () => ({
    LMS_BASE_URL: '<LMS_BASE_URL>',
    LOGOUT_URL: '<LOGOUT_URL>',
    SUPPORT_URL: '<SUPPORT_URL>',
  }),
}));
jest.mock('@edx/paragon', () => {
  const Dropdown = () => 'Dropdown';
  Dropdown.Toggle = () => 'Dropdown.Toggle';
  Dropdown.Menu = () => 'Dropdown.Menu';
  Dropdown.Item = () => 'Dropdown.Item';
  return { Dropdown };
});
jest.mock('@fortawesome/react-fontawesome', () => ({
  FontAwesomeIcon: () => 'FontAwesomeIcon',
}));
jest.mock('@fortawesome/free-solid-svg-icons', () => ({
  faUserCircle: 'fa-user-circle-icon',
}));

describe('Header AuthenticatedUserDropdown UserMenu component', () => {
  const props = {
    intl: { formatMessage: (msg) => msg.defaultMessage },
    username: 'test-username',
  };
  test('snapshot', () => {
    expect(
      shallow(<UserMenu {...props} />),
    ).toMatchSnapshot();
  });
});
