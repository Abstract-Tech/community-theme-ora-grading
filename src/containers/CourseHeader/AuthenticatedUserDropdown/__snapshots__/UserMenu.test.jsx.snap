// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header AuthenticatedUserDropdown UserMenu component snapshot 1`] = `
<Component
  className="dropdown-menu-right"
>
  <Component
    href="<LMS_BASE_URL>/dashboard"
  >
    Dashboard
  </Component>
  <Component
    href="<LMS_BASE_URL>/u/test-username"
  >
    Profile
  </Component>
  <Component
    href="<LMS_BASE_URL>/account/settings"
  >
    Account
  </Component>
  <Component
    href="<LOGOUT_URL>"
  >
    Sign Out
  </Component>
</Component>
`;
